android/app/src/main/java/hr/foi/air/wattsup/screens/CardScreen.kt

```diff
@@ -0,0 +1,366 @@
+import hr.foi.air.wattsup.network.models.TokenManager
+import hr.foi.air.wattsup.ui.component.CircleButton
+import hr.foi.air.wattsup.ui.component.TopAppBar
+import hr.foi.air.wattsup.viewmodels.CardViewModel
+
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun CardScreen(onArrowBackClick: () -> Unit, onAddCard: () -> Unit, viewModel: CardViewModel) {
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = { Text(stringResource(R.string.cards)) },
+                navigationIcon = {
+                    IconButton(onClick = { onArrowBackClick() }) {
+                        Icon(Icons.Filled.ArrowBack, null, tint = Color.White)
+                    }
+                },
+            )
+        },
+    ) {
+        CardView(viewModel, onAddCard)
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun CardView(viewModel: CardViewModel, onAddCard: () -> Unit, context: Context = LocalContext.current) {
+    val userId = TokenManager.getInstance(context).getId()
+    val state = rememberLazyListState()
+    val cards by viewModel.cards.observeAsState(emptyList())
+    val showLoading by viewModel.showLoading.observeAsState(true)
+    val scannedCard by viewModel.card.observeAsState()
+    val refresh = remember { mutableStateOf(false) }
+
+    AddDialog(refresh, scannedCard, userId, viewModel)
+
+    LaunchedEffect(Unit, refresh.value) {
+        viewModel.refreshCards(context, userId)
+    }
+
+    Column (
+        modifier = Modifier
+            .padding(top = 50.dp)
+            .fillMaxSize()
+    ) {
+        LazyRow(
+            modifier = Modifier
+                .fillMaxWidth()
+                .fillMaxHeight(0.85f),
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.Center,
+            state = state,
+            flingBehavior = rememberSnapFlingBehavior(lazyListState = state)
+        ) {
+            if (showLoading) {
+                item {
+                    CircularProgressIndicator(
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .wrapContentHeight(Alignment.CenterVertically)
+                    )
+                }
+            } else if (cards.isEmpty()) {
+                item {
+                    Text(
+                        text = "You have no cards",
+                        style = MaterialTheme.typography.headlineMedium,
+                        textAlign = TextAlign.Center,
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .wrapContentHeight(Alignment.CenterVertically)
+                    )
+                }
+            } else {
+                itemsIndexed(cards) { _, item ->
+                    CardCard(refresh, item!!, viewModel)
+                }
+            }
+        }
```

Comment:
Great implementation, for added effect I suggest switching from LazyRow to HorizontalPager in order to be able to track active card and add dots as indicators to see how many cards there currently are and on which card the user is at the moment. 

IndicatorRow could be something like:

```
@Composable
fun IndicatorRow(cardCount: Int, activeIndex: Int) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(top = 8.dp),
        horizontalArrangement = Arrangement.Center,
    ) {
        for (i in 0 until cardCount) {
            // Draw a dot for each card, with active card highlighted
            Box(
                modifier = Modifier
                    .size(8.dp)
                    .background(
                        if (i == activeIndex) MaterialTheme.colorScheme.secondary else Color.Gray,
                        RoundedCornerShape(4.dp),
                    )
                    .padding(10.dp),
            )
            Spacer(modifier = Modifier.width(10.dp))
        }
    }
}
```

Then if LazyRow is changed to HorizontalPager, IndicatorRow can be called below it:

```
// Initialize the pager state with the initial page index set to 0
    val pagerState = rememberPagerState(
        initialPage = 0,
        initialPageOffsetFraction = 0f,
    ) {
        cards.size
    }

    // Calculate the activeCardIndex based on the current page in the pager state
    val activeCardIndex = pagerState.currentPage

    Column(
        modifier = modifier
            .fillMaxSize(),
    ) {
        HorizontalPager(
            state = pagerState,
            modifier = Modifier
                .fillMaxWidth()
                .fillMaxHeight(0.6f),
            verticalAlignment = Alignment.CenterVertically,
        ) { page ->
            if (showLoading) {
                LoadingSpinner(
                    Modifier
                        .height(25.dp)
                        .width(25.dp)
                        .wrapContentSize(Alignment.Center),
                )
            } else if (cards.isEmpty()) {
                Text(
                    text = "You have no cards",
                    style = MaterialTheme.typography.headlineMedium,
                    textAlign = TextAlign.Center,
                    modifier = Modifier
                        .fillMaxSize()
                        .wrapContentHeight(Alignment.CenterVertically),
                )
            } else {
                CardCard(refresh, cards[page]!!, viewModel)
            }
        }
        IndicatorRow(cards.size, activeCardIndex)
```

(Additionally, CircularProgressIndicator was replaced with LoadingSpinner component which is basically the same indicator, but was extracted as a reusable component in `components` package.)

Reply:
I thought about it, but ended up finding a nice code snippet the indicator that works with LazyRow to avoid total refactoring. Changed spinner like mentioned.